src/Castle.Core.Tests/BasicClassProxyTestCase.cs

```diff
@@ -78,18 +78,15 @@ public void Caching()
 		}
 
 		[Test]
-#if __MonoCS__
-		[Ignore("Expected: Castle.DynamicProxy.Generators.GeneratorException, But was: System.ArgumentNullException")]
-#endif
 		public void ProxyForNonPublicClass()
 		{
 			// We need to use a type that is not from our assembly, because we are marked as internals visible to DynamicProxy2
-			var type = Type.GetType("System.__Canon"); // Don't specify the assembly name (it'll be either mscorlib or System.Private.CorLib)
+			var type = Type.GetType("NUnit.Framework.Internal.ActionsHelper, nunit.framework");
```

Comment:
Was this necessary, I'm surprised Mono wouldn't also have a matching `System.__Canon` class. The reason for my query is that `__Canon` will be very stable as an internal class where as that NUnit class might change/move in the future.

https://github.com/mono/mono/blob/73866f6b16bb09a34b3caa0271bb9fa6ae9d8331/mcs/class/referencesource/mscorlib/system/object.cs#L230-L237

Reply:
The Mono repo contains only very few [mentions of `__Canon`](https://github.com/mono/mono/search?q=__Canon). The file you linked to is one, but it appears to be relatively new. Also, I'm not sure why, but `System.__Canon` simply does not end up in Mono's `mscorlib.dll`, and I suppose that's why a call to `Type.GetType("System.__Canon")` returns `null` on Mono.

One notable internal type that both runtimes have is `System.__ComObject`. However, if you try to proxy that even on the CLR, you'll get a `FatalExecutionEngineError`. 😆

Not sure what to do about this. Should the test just be disabled on Mono?

Reply:
Nice find! :+1: I've updated the test accordingly. Works fine!